### PR TITLE
Unify paths for epoll

### PR DESCRIPTION
**Pull request by rennergade**
epoll_create
epoll_ctl
epoll_wait

### CODE REVIEW

**Review by kapkic — APPROVED** [review]
LGTM

File: src/shared/platform/lind_platform.c [hunk 1]
```diff
@@ -764,10 +764,37 @@ int lind_gethostname (char *name, size_t len, int cageid)
...
+    callArgs = Py_BuildValue("(i[ii])", LIND_safe_net_epoll_create, size, cageid);
+    LIND_API_PART2;
+    LIND_API_PART3;
+}
+
+int lind_epoll_ctl(int epfd, int op, int fd, struct epoll_event *event, int cageid)
+{
+    LIND_API_PART1;
+    callArgs = Py_BuildValue("(i[iiii])", LIND_safe_net_epoll_ctl, epfd, op, fd, cageid);
+    COPY_DATA(event, sizeof(*event))
```
Comment (jesings) [inline src/shared/platform/lind_platform.c]: COPY_DATA has to be after LIND_API_PART2

File: src/shared/platform/lind_platform.c [hunk 2]
```diff
@@ -764,10 +764,37 @@ int lind_gethostname (char *name, size_t len, int cageid)
...
+    LIND_API_PART2;
+    LIND_API_PART3;
+}
+
+int lind_epoll_wait(int epfd, struct epoll_event *events, int maxevents, int timeout, int cageid)
+{
+    LIND_API_PART1;
+    callArgs = Py_BuildValue("(i[iiii])", LIND_safe_net_epoll_wait, epfd, maxevents, timeout, cageid);
+    COPY_DATA(events, sizeof(*events))
```
Comment (jesings) [inline src/shared/platform/lind_platform.c]: Same comment as above, move this down one line

File: src/trusted/service_runtime/lind_syscalls.c [hunk 3]
```diff
@@ -616,9 +534,9 @@ StubType stubs[] = {
         {0}, /* 53 */
         {0}, /* 54 */
         {0}, /* 55 */
-        {LindEpollCreatePreprocess, LindEpollCreatePostprocess, 0}, /* 56 epoll_create */
-        {LindEpollCtlPreprocess, 0, 0}, /* 57 epoll_ctl */
-        {LindEpollWaitPreprocess, LindEpollWaitPostprocess, 0}, /* 58 epoll_wait */
+        {0}, /* 56 epoll_create */
+        {0}, /* 57 epoll_ctl */
+        {0}, /* 58 epoll_wait */
```
Comment (jesings) [inline src/trusted/service_runtime/lind_syscalls.c]: It also might be good to remove these comments, I've been trying to do so to keep track of what we've gotten rid of (calls that don't need any preprocessing etc. will just have their name commented and still have the array entry be {0}

File: src/trusted/service_runtime/nacl_syscall_common.c [hunk 4]
```diff
@@ -5066,3 +5075,123 @@ int32_t NaClSysFcntlSet (struct NaClAppThread *natp,
...
+              NaClDescSafeUnref(ndp);
+              continue;
+          }
+          hfd = ((struct NaClDescIoDesc *)ndp)->hd->d;
+          if(pfds[i].data.fd == hfd) {
+              pfds[i].data.fd = j;
+          }
+          NaClDescUnref(ndp);
+      }
+  }
```
Comment (jesings) [inline src/trusted/service_runtime/nacl_syscall_common.c]: This is really ugly but I guess it's the only way this can be accomplished.

**Review by jesings — COMMENTED** [review]
Just a couple of things to take care of, overall seems good!

**Comment by rennergade** [comment]
I think I fixed all those thanks!

**Review by jesings — APPROVED** [review]

**Pull request merged**